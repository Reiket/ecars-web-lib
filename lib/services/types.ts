export interface ElementProps {
  block?: string;
}

export interface AccordionItem {
  title: string;
  text: string;
}
