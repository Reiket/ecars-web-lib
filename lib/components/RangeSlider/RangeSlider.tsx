import type {FC} from 'react';
import type {SliderProps} from 'rc-slider';
import Slider from 'rc-slider';

export const RangeSlider: FC<SliderProps> = (props) => {
  return (
    <Slider
      {...props}
      range
    />
  );
};
