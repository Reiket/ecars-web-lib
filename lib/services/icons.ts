import {MdOutlineKeyboardArrowLeft, MdOutlineKeyboardArrowRight} from 'react-icons/md';
import {FaCheck} from 'react-icons/fa6';

export const Icons = {
  ArrowNarrowRight: MdOutlineKeyboardArrowRight,
  ArrowNarrowLeft: MdOutlineKeyboardArrowLeft,
  Check: FaCheck,
};
