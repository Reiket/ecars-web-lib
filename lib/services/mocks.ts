export const selectOptionsMock = ['Option 1', 'Option 2', 'Another Option'];
export const totalCountPageMock = 100;
export const pageSizeMock = 10;

export const imagesMock = Array.from({length: 6}, (_) => ({
  original: 'src/assets/react.svg',
  thumbnail: 'src/assets/react.svg',
}));

export const dropdownOptionsMock = [
  {
    value: '1',
  },
  {
    value: '2',
  },
];
