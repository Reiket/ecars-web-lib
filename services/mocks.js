const o = ["Option 1", "Option 2", "Another Option"];
export {
  o as selectOptionsMock
};
//# sourceMappingURL=mocks.js.map
