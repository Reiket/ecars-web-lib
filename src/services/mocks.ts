export const accordionContentMock = [
  {
    title: 'What are the guarantees?',
    text: "At ECars, we understand the importance of assurance when investing in our products. That's why we offer a comprehensive range of guarantees to ensure your satisfaction and peace of mind.",
  },
  {
    title: 'How much does shipping cost?',
    text: 'Shipping varies depending on the car you are buying as its calculated by the dimensions of each unit. There is RORO and Container so depending on your choice we will be able to give you an estimate, closer to the shipping date we will share an exact quote.',
  },
];
