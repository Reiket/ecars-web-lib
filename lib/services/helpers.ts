import type {ClassValue} from 'clsx';
import {clsx} from 'clsx';
import {Icons} from '@/services/icons';
import type {ComponentType} from 'react';

export const cn = (blockName = '', elementName: string, ...inputs: ClassValue[]): string => {
  const className = blockName ? `${blockName}__${elementName} ${elementName}` : elementName;
  return clsx(className, inputs);
};

export const capitalizeFirstLetter = (string: string): string => {
  return string.charAt(0).toUpperCase() + string.slice(1);
};

interface PasswordFieldPropsResult {
  inputType: string;
  ButtonIcon: ComponentType;
}

export const getPasswordFieldProps = (isVisible: boolean): PasswordFieldPropsResult => ({
  inputType: isVisible ? 'text' : 'password',
  ButtonIcon: isVisible ? Icons.OpenEye : Icons.CloseEye,
});
