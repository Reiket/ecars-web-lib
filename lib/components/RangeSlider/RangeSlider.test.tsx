import React from 'react';
import {render, screen} from '@testing-library/react';
import {RangeSlider} from './RangeSlider';
import '@testing-library/jest-dom';
import {rangeSliderArrayMock} from '@/services/mocks';

describe('RangeSlider Component', () => {
  test('renders without crashing', () => {
    const {container} = render(<RangeSlider defaultValue={rangeSliderArrayMock} />);
    const sliderContainer = container.querySelector('.rc-slider');
    expect(sliderContainer).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('renders with correct default range', () => {
    render(<RangeSlider defaultValue={rangeSliderArrayMock} />);
    const thumbs = screen.getAllByRole('slider', {hidden: true});
    expect(thumbs).toHaveLength(2);
  });
});
