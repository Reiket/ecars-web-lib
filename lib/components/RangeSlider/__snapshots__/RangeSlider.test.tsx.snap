// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSlider Component renders Range SliderComponent component correctly 1`] = `
<div>
  <div
    class="rc-slider range-slider rc-slider-horizontal"
  >
    <div
      class="rc-slider-rail"
    />
    <div
      class="rc-slider-track rc-slider-track-1"
      style="left: 10%; width: 90%;"
    />
    <div
      class="rc-slider-step"
    />
    <div
      aria-disabled="false"
      aria-orientation="horizontal"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="10"
      class="rc-slider-handle rc-slider-handle-1"
      role="slider"
      style="left: 10%; transform: translateX(-50%);"
      tabindex="0"
    />
    <div
      aria-disabled="false"
      aria-orientation="horizontal"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="100"
      class="rc-slider-handle rc-slider-handle-2"
      role="slider"
      style="left: 100%; transform: translateX(-50%);"
      tabindex="0"
    />
  </div>
</div>
`;
